src/main/drivers/system_stm32f4xx.c

```diff
@@ -39,18 +39,39 @@ void systemReset(void)
     NVIC_SystemReset();
 }
 
+PERSISTENT uint32_t bootloaderRequest = 0;
+
 void systemResetToBootloader(void)
 {
     if (mpuResetFn) {
         mpuResetFn();
     }
 
-    *((uint32_t *)0x2001FFFC) = 0xDEADBEEF; // 128KB SRAM STM32F4XX
+    bootloaderRequest = 1;
 
     __disable_irq();
     NVIC_SystemReset();
 }
 
+#define SYSTEM_MEMORY __attribute__((section(".system_isr_vector"), aligned(4)))
+typedef void resetHandler_t(void);
+
+// System memory interrupt vector table
+static SYSTEM_MEMORY uint32_t           const volatile  systemStackEnd;
```

Comment:
IMO this is very nonportable - there is no guarantee that variables will end up in correct order (and place) in this section. (I assume that) It also overlaps existing memory region with actual vectors ...
Just use `#define`s, in similar way as STM headers define them ...

You may also define `struct` and cast 0x1FFF0000 to it .. 

Also you can define symbols in linker script, if you want to put address there